index.js

```diff
@@ -459,32 +487,59 @@ function flameGraph (opts) {
       context.fill()
     }
 
-    if (width >= 35) {
-      context.save()
-      context.clip()
-      context.font = '12px Verdana'
-      context.fillStyle = '#000'
-
-      // Center the "all stacks" text
-      if (!node.parent) {
-        context.textAlign = 'center'
-        x += width / 2
-      } else {
-        x += 4 // add padding to other nodes
-      }
-
-      var label = labelName(node)
-      context.fillText(label, x, h - (depth * c) - 1)
+    if (heatBars) {
+      context.beginPath()
+      context.moveTo(x, y)
+      context.lineTo(x, y + c)
+      context.moveTo(x + width, y)
+      context.lineTo(x + width, y + c)
+      context.stroke()
+    } else {
+      context.stroke()
+    }
+  }
 
-      var stack = labelStack(node)
-      if (stack) {
-        var offs = context.measureText(label + ' ').width
-        context.font = '10px Verdana'
-        context.fillText(stack, x + offs, h - (depth * c) - 2)
-      }
+  function renderLabel (context, node, x, y, width) {
+    context.save()
+    context.beginPath()
+    context.rect(x, y, width, c)
+    context.clip()
+    context.font = c > 20 ? `16px ${fontFamily}` : `12px ${fontFamily}`
+    context.fillStyle = labelColors[node.data.type] || labelColors.default
+
+    var labelOffset = 4 // padding
+    // Center the "all stacks" text
+    if (!node.parent) {
+      context.textAlign = 'center'
+      labelOffset = width / 2
+    }
 
-      context.restore()
+    // Magic value to sorta kinda align the label in the middle of the frame height
+    // It's not very accurate
+    var btmOffset = Math.floor((c - 16) / 2)
+    var label = labelName(node)
+    context.fillText(label, x + labelOffset, y + c - btmOffset)
```

Comment:
I'd like to revisit all of this and calculate the font size relative to the cell height, but it's not something we need to spend time on right now.

Reply:
Yea, I tried to simply scale it linearly but it needs a slightly less simplistic approach. At lower heights, the font size should be relatively large to be readable, but at higher heights, it should be relatively small to have some breathing room